hydra-node/exe/hydra-node/Main.hs

```diff
@@ -110,6 +126,29 @@ main = do
 
     action (Ledger.cardanoLedger globals ledgerEnv)
 
+  -- check if hydra-node parameters are matching with the hydra-node state.
+  -- REVIEW: Should we also check against items we receive on-chain here? e.g. 'OpenThreadOutput' if in Open state?
+  checkParamsAgainstExistingState :: HeadState Ledger.Tx -> Environment -> [String]
```

Comment:
I would probably return a list of some error type here instead of a list of string and just post-pone the error string formatting later. That would probably make the error management code more explicit.
```suggestion
  checkParamsAgainstExistingState :: HeadState Ledger.Tx -> Environment -> [IncompatibleParameter]
```

Reply:
definitely agree